### ISSUE TITLE
[hotfix/unauth_access_frontend] Unauthorized access to frontend due to href and goto on public pages

### ISSUE THREAD

**Issue by GeneralMine** [GeneralMine/S2QUAT#9]
You can gain access to the whole frontend by clicking here https://s2quat.raiser.dev/impressum on the logo.
So when ever a `href` or `goto` to "/" is executed within public pages, the sessionChecker gets bypassed and the frontend gets accessible. 

Due to the caching of svelte/sapper?

@ciearius need your help here

**Comment by GeneralMine** [comment]
At least the backend detects the unauthorized access and rejects the request. So the data is protected

**Comment by ciearius** [comment]
Did not manage to recreate this btw.

As far as caching is concerned, this is expected behavior. We should take a look at the redirection away from protected pages if no session is provided tho.
We did have redirection logic, right?

fyi: 
official svelte discord
<img width="1233" alt="image" src="https://user-images.githubusercontent.com/38101391/108629568-d5f1ad00-7460-11eb-8029-c1cf35f9eb38.png">

sapper docs:
https://sapper.svelte.dev/docs#Pages

tl;dr
In achieving an app like feel we lose the ability to hide frontend code, since it's all sent to the client in the initial load (with some exceptions I think).

Super tl;dr
Nothing sensitive or confidential should be hardcoded in the frontend. 
We have to work on the redirection tho.

**Comment by GeneralMine** [comment]
To sum up:
- the access to the database and all the data is still secure!
- theoretical we cannot restrict the access to the frontend GUI entirely. The reason lays in the frontend technology (Svelte) -> its only cosmetics and gets always auto cached. Everyone who "breaks in" can only see the buttons and the layout, but the page will be empty due to the backend security
- To solve the issue that everyone can **easily** get access to the frontend we just have to remove all public redirects from "/" to "/login"

**Comment by GeneralMine** [comment]
already fixed
